internal-dns/src/resolver.rs

```diff
@@ -55,15 +55,15 @@ impl Resolver {
     /// Construct a new DNS resolver from specific DNS server addresses.
     pub fn new_from_addrs(
         log: slog::Logger,
-        dns_addrs: Vec<SocketAddr>,
+        dns_addrs: &[SocketAddr],
     ) -> Result<Self, ResolveError> {
         info!(log, "new DNS resolver"; "addresses" => ?dns_addrs);
 
         let mut rc = ResolverConfig::new();
         let dns_server_count = dns_addrs.len();
         for socket_addr in dns_addrs.into_iter() {
             rc.add_name_server(NameServerConfig {
-                socket_addr,
+                socket_addr: *socket_addr,
```

Comment:
This is a really silly nit, but if you find this unsightly, I think you can do

```rust
for &socket_addr in dns_addrs.into_iter() {
```

to avoid needing the deref here (since `SocketAddr` impls `Copy`).